### ISSUE TITLE
installation of AmberTools v20 is broken due to switch to CMake in custom easyblock for Amber

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-easyconfigs#14021]
In https://github.com/easybuilders/easybuild-easyblocks/pull/2445 we updated the Amber easyblock to use CMake, but we failed to take into account that that same easyblock is also used for `AmberTools`.

Problems with `AmberTools-20-intel-2020a-Python-3.8.2.eb`:

* the runtime patching done on `AmberTools/src/cpptraj/configure` is failing, because the relative path is now run (building is done from a separate build dir due to the switch to `CMakeMake`), can be fixed by using `%(builddir)s/AmberTools/src/cpptraj/configure` (but not sure if that patching is still needed when using CMake)
* CMake is missing as a build dependency
* configuring with CMake fails with:
```
CMake Error at cmake/LibraryTracking.cmake:174 (message):
  Incorrect usage.  At least one LIBRARY should be provided.
Call Stack (most recent call first):
  cmake/MPIConfig.cmake:122 (import_libraries)
  CMakeLists.txt:118 (include)
  ```

Easy fix may be to tweak the `Amber` easyblock a bit more to stick to the old approach for `AmberTools`?

@branfosj @akesandgren Thoughts?

**Comment by branfosj** [comment]
I'd prefer switching to cmake for ambertools as well.

Adding https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/Amber/AmberTools-20_fix_missing_MPI_LIBRARY_error.patch should fix the cmake fail.

**Comment by akesandgren** [comment]
Yeah that patch is vital, and any other AmberTools patches from Amber should be added to the AmberTools build.
I do try to make sure that the patches in Amber are correctly named, i.e. AmberTools vs Amber.
So these should be part of AmberTools:
```
AmberTools-20_cmake-locate-netcdf.patch
AmberTools-20_fix_missing_MPI_LIBRARY_error.patch
AmberTools-20_fix_xblas_missing_make_dependency.patch
```

**Comment by boegel** [comment]
Thanks, that patch helps.

Now I'm hitting this:

```
CMake Error at cmake/PythonBuildConfig.cmake:67 (message):
  Missing required Python packages: numpy scipy matplotlib.  Please install
  these and try again.  If you cannot install them, you may set BUILD_PYTHON
  to FALSE to skip building Python packages, or set DOWNLOAD_MINICONDA to
  TRUE to create a python environment automatically.
```

So the CMake-based installation is building more stuff, leading to more dependencies? Ugh...

**Comment by akesandgren** [comment]
See Amber for deps :-) the diff in deps between them is basically zero

**Comment by boegel** [comment]
OK, but why does `AmberTools` require more dependencies when building with CMake compared to without it?

I had to add `SciPy-bundle` and `matplotlib`, but now also `FFTW` (since the existing easyconfig for `AmberTools` is built with `intel` toolchain):

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
FFTW_LIBRARY_SERIAL
    linked by target "cmTC_da004" in directory /tmp/easybuild_build/AmberTools/20/intel-2020a-Python-3.8.2/easybuild_obj/CMakeFiles/CMakeTmp

CMake Error at cmake/TryLinkLibrary.cmake:135 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  cmake/jedbrown/FindFFTW.cmake:207 (try_link_library)
  cmake/3rdPartyTools.cmake:240 (find_package)
  CMakeLists.txt:194 (include)
```

**Comment by akesandgren** [comment]
Because the list wasn't complete before?
Haven't done an intel build of Amber lately so I don't know the deps for that.

And the CMake based build is much better at searching and finding external deps.

**Comment by boegel** [comment]
Made some progress in #14028, but not finished yet, running into a problem with `sander`, not sure how to replace the `-nosanderapi` configuration option with an equivalent that CMake listens to...

Any input welcome!

**Comment by branfosj** [comment]
Possibly `BUILD_SANDER_API`. If not then I'll look further when I am next attached to a system with the Amber tools source code.
